src/corelibs/U2Core/src/dbi/U2DbiPackUtils.cpp

```diff
@@ -295,15 +264,15 @@ QByteArray U2DbiPackUtils::packRow(qint64 posInMca, const U2McaRow &row) {
     return result;
 }
 
-bool U2DbiPackUtils::unpackRow(const QByteArray &modDetails, qint64 &posInMca, U2McaRow &row) {
+bool U2DbiPackUtils::unpackRow(const QByteArray &modDetails, int &rowIndex, U2McaRow &row) {
     QList<QByteArray> tokens = modDetails.split(SEP);
-    SAFE_POINT(8 == tokens.size(), QString("Invalid added row modDetails string '%1'").arg(QString(modDetails)), false);
+    SAFE_POINT(tokens.size() == 8, QString("Invalid added row modDetails string '%1'").arg(QString(modDetails)), false);
     {  // version
         SAFE_POINT(VERSION == tokens[0], QString("Invalid modDetails version '%1'").arg(tokens[0].data()), false);
     }
-    {  // posInMsa
+    {  // row index
         bool ok = false;
-        posInMca = tokens[1].toLongLong(&ok);
+        rowIndex = tokens[1].toInt(&ok);
```

Comment:
You may lose the data here if `tokens[1]` is large enought

Reply:
How the tokens[1] can be large enough here?